napari/utils/io.py

```diff
@@ -48,3 +48,27 @@ def imsave(filename: str, data: np.ndarray):
         import imageio
 
         imageio.imsave(filename, data)
+
+
+def __getattr__(name: str):
+    if name in {
+        'imsave_extensions',
+        'write_csv',
+        'read_csv',
+        'csv_to_layer_data',
+        'read_zarr_dataset',
+    }:
+        warnings.warn(
+            trans._(
+                '{name} was moved from napari.utils.io in 0.4.17. Import it from napari_builtins.io instead.',
+                deferred=True,
+                name=name,
+            ),
```

Comment:
As it is pure for plugin developers, I think it should not be translated. If a user sends me a screenshot with such information, I prefer it to be searchable.

Reply:
Based on [the napari translation docs](https://napari.org/stable/developers/translations.html#deferred-translations), I think `deferred=True` gives us a nice compromise - the translated message will be shown in the notification window, but the original (likely English) message is shown in any terminal/log for better debugging/searching. So I'm going to leave this one as is for now.